src/qtt/algorithms/functions.py

```diff
@@ -337,20 +339,38 @@ def fit_gauss_ramsey(x_data, y_data, weight_power=0, maxiter=None, maxfun=5000,
 
     """
 
-    def func(params): return cost_gauss_ramsey(x_data, y_data, params, weight_power=weight_power)
+    def gauss_ramsey_model(x, amplitude, decay_time, frequency, phase, offset):
+        """  """
+        y = gauss_ramsey(x, [amplitude, decay_time, frequency, phase, offset])
+        return y
+
+    if weight_power is None:
+        weights = None
+    else:
+        diff_x = np.diff(x_data)
+        weights =  np.hstack( (diff_x[0], diff_x) )  ** weight_power
```

Comment:
Too many spaces. I understand you put in some spaces to indicate we are dealing with a tuple instead of 2 parameters. Nice.
```suggestion
        weights = np.hstack( (diff_x[0], diff_x) ) ** weight_power
```